clients/src/main/java/org/apache/kafka/common/protocol/Errors.java

```diff
@@ -495,7 +496,14 @@ public ApiException build(String message) {
             public ApiException build(String message) {
                 return new OperationNotAttemptedException(message);
             }
-        });
+    }),
+    KAFKA_STORAGE_ERROR(56, "Disk error when trying to access log file on the disk.",
```

Comment:
@junrao @lindong28 Yes, my suggestion was to bump the protocol so that we reuse an existing retriable exception for the old version and `KafkaStorageException` for the new version. And `RetriableUnknownException` would be a way to avoid this hack in the future, but it doesn't help now.